### ISSUE TITLE
Pos objects preventing Garbage Collection of views

### ISSUE THREAD

**Issue by a-usr** [gui-cs/Terminal.Gui#2913]
**Is your feature request related to a problem? Please describe.**
As the Title suggests, I have a problem with the fact that Views hold on to their pos objects even after calling `IDisposable.Dispose`.
Because of the nature of `Pos` objects, this prevents the view from being collected by GC. This would normally not be much of an issue, but, as i am dealing with Assembly Loading and unloading, it is a very important requirement for an object to be able to be GC'd immediatly after calling Dispose, as else the Assembly containing the object will not be able to be unloaded when it is expected to be.
However, as there probably aren't too many other people that have to deal with this, i have another reason why this should be resolved: Performance. When creating a Tui Application of large size, you sometimes have to discard large objects that you don't need at the moment, but can easily be rebuilt. This logic however doesn't apply if the object you dereferenced from your program is still fixed in ram by some static variable. 

**Describe the solution you'd like**
I think the solution to this is actually relatively simple: Alter the `Dispose` implementation in `base.View.cs` so that it dereferences (Or maybe even destructs) the `Pos` and `Dim` objects stored in the `x`, `y`, `width`, `height`, and any other variable that may or may not store a `Pos` or `Dim` object. Simple as that (At least i think so, so please correct me if I'm wrong).

**Describe alternatives you've considered**
Creating an override of the `View` class and forcing coworkers to use it instead of the base. But the id have to re-Implement the child classes of `View` (the base). 

**Additional context**
I might create a Pull Request to solve this myself, but only once i have confirmation that this actually would work

**Comment by tznind** [comment]
Surely after all references to a `View` are gone then all sub property references will also be free for GC?

Maybe there is a lingering reference to the `View` itself? e.g. via an event registration or something.

Could be related to https://github.com/gui-cs/Terminal.Gui/issues/2888 where an internal artifact of the `View` (Autocomplete) is still sticking around.

Are you using v1 or v2?

**Comment by a-usr** [comment]
> Surely after all references to a `View` are gone then all sub property references will also be free for GC?
Nope. After Some Debugging (Inspecting GC Roots using vs 2022 Debugger) I came to the conclusion that a Label was holding a reference to a Pos Object, whose posCombine Field was holding a reference to a custom View in the same superview. (This being the result of Positioning the label below the custom View using `Pos.Bottom` Computed Positioning)
> Maybe there is a lingering reference to the `View` itself? e.g. via an event registration or something.
Nope.
> Could be related to #2888 - where an internal artifact of the `View` (Autocomplete) is still sticking around.
Dont think so
> Are you using v1 or v2?
I am using v1

Also, worth noting is that after using a local build that had the changes i recommended, it worked just as i expected it to. I'll make two new PR's in a bit, and will link them here.

**Comment by a-usr** [comment]
I have made i PR solving this issue:
[HERE](https://github.com/gui-cs/Terminal.Gui/pull/2915)
I made another PR to solve Debugging issues i had when trying to unload an assembly with Terminal.Gui compiled in Debugging mode. I am not sure wether responder.Instances is supposed to hold onto instances even asfter they are disposed, wich is why this [Pull Request](https://github.com/gui-cs/Terminal.Gui/pull/2914) is optional.

**Comment by a-usr** [comment]
The Reason for the GC not releasing the object is that posCombine and many other fields inside Pos and Dim are statics. 

And, as statics are stored not inside of instance, but in the type itself (at least from my understanding) the GC refuses to release an object that is referenced inside a static variable.

Statics are cool, but only if you dont have to worry about recycling objects.

**Comment by BDisp** [comment]
> The Reason for the GC not releasing the object is that posCombine and many other fields inside Pos and Dim are statics.
> 
You are right about this. The static helper methods are great to automate instance creation of Pos/Dim objects because they really create a new instance, but the static fields objects are wrong because they will always hold an instance.

> And, as statics are stored not inside of instance, but in the type itself (at least from my understanding) the GC refuses to release an object that is referenced inside a static variable.
> 
I think you are right. If an object was set to null it will also hold it as null.

> Statics are cool, but only if you dont have to worry about recycling objects.
> 
Or if we want to recicle objects and use them to creating an instance without explicit it. That what the static methods is really doing.

> Also why are posCombine etc static anyways? I cant seem to understand why, and how their values seem to differ from instance to instance even though their static?
> 
Yes, it's really unnecessary because they represent the instance itself.

> Are they newly assigned each time the program needs the position of the view?

If it's that I don't think it needed to be store on a static field but on an instance field.
